src/composables/notifications.ts

```diff
@@ -0,0 +1,227 @@
+import {
+  computed, onBeforeUnmount, onMounted, ref,
+} from '@vue/composition-api';
+import {
+  IAccount, IChainNames,
+  INetwork,
+  INotification,
+  INotificationSetting,
+  NotificationStatus,
+} from '../types';
+import { useGetter, useState } from './vuex';
+import {
+  NOTIFICATION_STATUS_CREATED,
+  NOTIFICATION_STATUS_READ,
+  NOTIFICATION_TYPE_WALLET,
+  executeAndSetInterval,
+  watchUntilTruthy, fetchJson, postJson, NOTIFICATION_ENTITY_TYPE_TIP, AGGREGATOR_URL,
+} from '../popup/utils';
+
+const fetchedNotificationsLimit = 20;
+
+const mountedComponentsCounter = ref<number>(0);
+const notificationsSuperhero = ref<INotification[]>([]);
+const notificationsWallet = ref<INotification[]>([]);
+let pollingInterval: NodeJS.Timer | null = null;
+
+export function useNotifications(preventMountedIncrement = false) {
+  const sdk = useGetter('sdkPlugin/sdk');
+  const activeNetwork = useGetter<INetwork>('activeNetwork');
+  const account = useGetter<IAccount>('account');
+  const notificationSettings = useState<INotificationSetting[]>('notificationSettings');
+
+  const canLoadMore = ref(true);
+  const fetchedNotificationsOffset = ref(0);
+
+  const notificationSettingsCheckedTypes = computed<string[]>(
+    () => notificationSettings.value
+      .filter((setting: INotificationSetting) => setting.checked)
```

Comment:
No need to use `INotificationSetting` as the `setting`'s type is inherited from the `notificationSettings`.

You can also go with:
```
.filter(({ checked }) => checked)
```